### PR TITLE
Add support for (hover) 'text' in mesh3d traces

### PR DESCRIPTION
**Pull request by etpinard**
Call this a bug fix or a new feature, this PR closes https://github.com/plotly/plotly.js/issues/2242

### CODE REVIEW

**Comment by alexcjohnson** [comment]
Not sure how much you want to let this PR expand, but here's a definitely-a-bug that popped up as I was trying out your PR: `hoverinfo` has the wrong `editType`:
`Plotly.restyle(gd,{hoverinfo:'x+y'})`
does not update the displayed hover info for mesh3d. If you follow it with
`Plotly.redraw(gd)`
or include something else in the restyle though, it does, like
`Plotly.restyle(gd,{hoverinfo:'x+text',text:'AAAAAAAH'})`

Also it looks like `surface` doesn't support `text` as a single string (only as an array) and both `surface` and `scatter3d` have the same `hoverinfo` restyle problem as `mesh3d`.

**Comment by etpinard** [comment]
@alexcjohnson good catch. Thanks.

I think I fixed them all in https://github.com/plotly/plotly.js/pull/2327/commits/8ae2d0e0acc669f8e097cc4e9353e85e3e34ed1c by overriding the `hoverinfo` attribute declaration with `editType: 'calc'`.

**Comment by alexcjohnson** [comment]
https://github.com/plotly/plotly.js/commit/8ae2d0e0acc669f8e097cc4e9353e85e3e34ed1c looks great, thanks for the tests 🔒 
Do you want to extend `surface` to support `text` as a single string now too, so it matches the new `mesh3d.text`, or leave that in an issue?

**Comment by etpinard** [comment]
> Do you want to extend surface to support text as a single string now too

Oops. I pushed too soon. That's in https://github.com/plotly/plotly.js/pull/2327/commits/d20bdc1aae4e7ec42e27678c78951c429687b9ca

File: src/traces/surface/attributes.js [hunk 1]
```diff
@@ -113,9 +113,18 @@ var attrs = module.exports = overrideAll({
     },
 
     text: {
-        valType: 'data_array',
-        description: 'Sets the text elements associated with each z value.'
+        valType: 'string',
+        role: 'info',
+        dflt: '',
+        arrayOk: true,
+        editType: 'calc',
```
Comment (alexcjohnson) [inline src/traces/surface/attributes.js]: Oh this is in an `overrideAll` so you don't need `editType` - which I think applies to a few more of the additions in this PR too.

Comment (alexcjohnson) [reply src/traces/surface/attributes.js]: (I see you 🔪 some `editType` lines, just not the one I made the comment directly on 😅 )

Comment (etpinard) [reply src/traces/surface/attributes.js]: Oops. Fixed in -> 13a3ecf

**Comment by alexcjohnson** [comment]
Nice - slowly getting all these trace types up to par with each other. 💃

**Pull request merged**